### PR TITLE
feat(consumers): rust consumers quantized rebalance

### PR DESCRIPTION
**Pull request by volokluev**
Add the ability to do quantized rebalancing for consumers, toggleable by runtime config.

### What is quantized rebalancing?

Rebalancing events are synchronized to the tick of the clock. Controlled by the runtime config: `quantized_rebalance_consumer_group_delay_secs__{consumer_group}`

Let's say: 

`SET  quantized_rebalance_consumer_group_delay_secs__spans=15` 

* When a consumer starts up: delay subscribing to a topic until `timestamp % 15 == 0`
* when a consumer gets a shutdown signal (ctrl-c): delay leaving the group until `timestamp % 15 == 0`

### What else is in this PR?

This PR adds the ability to read from redis config directly. This was mostly done because I couldn't get the python bindings to work on my machine, it will still fall back to the old implementation if there is an error in the direct implementation

### CODE REVIEW

File: rust_snuba/src/consumer.rs [hunk 1]
```diff
@@ -286,10 +292,21 @@ pub fn consumer_impl(
...
 
-    ctrlc::set_handler(move || {
-        handle.signal_shutdown();
-    })
-    .expect("Error setting Ctrl-C handler");
+    match rebalance_delay_secs {
+        Some(secs) => {
+            ctrlc::set_handler(move || {
+                rebalancing::delay_kafka_rebalance(secs);
+                handle.signal_shutdown();
```
Comment (john-z-yang) [inline rust_snuba/src/consumer.rs:299]: Am I correct in assuming that this closure that is passed in runs in a separate thread, so the main loop is not blocked while it is sleeping before signaling shutdown?

**Comment by john-z-yang** [comment]
This seems reasonable to me, do we plan to have quantized rebalancing on the other end? I.e. Making the consumer wait a little bit before joining the consumer group

File: rust_snuba/src/consumer.rs (hunk 1 above)
Comment (john-z-yang) [inline rust_snuba/src/consumer.rs:299]: I'm curious if arroyo let's us to the following
-

Comment (volokluev) [reply rust_snuba/src/consumer.rs:299]: From the documentation it seems to be the case

https://docs.rs/ctrlc/latest/ctrlc/fn.set_handler.html

```
Register signal handler for Ctrl-C.

Starts a new dedicated signal handling thread. Should only be called once, typically at the start of your program.
```

**Comment by volokluev** [comment]
@john-z-yang it does wait before joining, see

https://github.com/getsentry/snuba/pull/6561/files#diff-03eab290aeb86b49a5a13b44a3a48fd7cae4efdc4ae908e4e92309f8241aaac9R238-R242

(and I tested that this happens locally as well)

File: rust_snuba/src/rebalancing.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    ) {
+        Ok(delay_secs) => match delay_secs {
+            Some(secs) => match secs.parse() {
+                Ok(v) => Some(v),
+                Err(_) => None,
+            },
+            None => None,
+        },
+        Err(_) => None,
+    }
```
[the remark this answers is not in the document]
Comment (volokluev) [reply rust_snuba/src/rebalancing.rs:47]: nice, today I learned about `.ok` :D thank you john

**Review by john-z-yang — APPROVED** [review]

File: rust_snuba/src/consumer.rs (hunk 1 above)
Comment (john-z-yang) [reply rust_snuba/src/consumer.rs:299]: Nice!

**Pull request merged**